source/common/http/conn_manager_impl.cc

```diff
@@ -1468,6 +1469,30 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ResponseHeaderMap& heade
 
   chargeStats(headers);
 
+  if (auto* proxy_status_config = connection_manager_.config_.proxyStatusConfig();
```

Comment:
It makes sense to move this into a dedicated method and then call it from the `ConnectionManagerUtility::mutateResponseHeaders` to keep all the response header mutations in one place.

Reply:
Done. I made a new method named ConnectionManagerUtility::setProxyStatusHeader., then called that method from ConnectionManagerUtility::mutateResponseHeaders. This necessitated adding two new arguments to ConnectionManagerUtility::mutateResponseHeaders and propagating that change to all callsites and test callsites, but it seemed like the right move -- understood that it's best to keep all response header mutation together.